src/coreneuron/io/nrn_setup.cpp

```diff
@@ -931,33 +943,84 @@ void read_phase3(NrnThread& nt, UserParams& userParams) {
 
     /** mapping information for all neurons in single NrnThread */
     NrnThreadMappingInfo* ntmapping = new NrnThreadMappingInfo();
+    if (corenrn_embedded) {
```

Comment:
The read_phase3() function is a bit big now...
Could you split it in at least 2 functions? For the in memory and from file read

Maybe something like:

```
void read_phase3(NrnThread& nt, UserParams& userParams) {
    auto& F = userParams.file_reader[nt.id];
    F.restore_checkpoint();
    NrnThreadMappingInfo* ntmapping = new NrnThreadMappingInfo();
    if (corenrn_embedded) {
        read_mapping_info(ntmapping);
    } else {
        read_mapping_info_from_file(ntmapping, F);
    }
    ...
}
```